numpy-stubs/__init__.pyi

```diff
@@ -919,3 +920,69 @@ def argsort(
     kind: Optional[_SortKind] = ...,
     order: Union[None, str, Sequence[str]] = ...,
 ) -> ndarray: ...
+@overload
+def argmax(
+    a: Union[Sequence[ArrayLike], ndarray],
```

Comment:
Why not `ArrayLike` here and below?

Reply:
So the likes of `argmax()`, `argmin()` and `diagonal()` are one of the few cases where, at the very least, a vector or matrix is required. Hence the whole `Sequence[ArrayLike]` and `Sequence[Sequence[ArrayLike]]` syntaxes.